examples/chip-tool/commands/icd/ChipToolICDRegistrationDelegate.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "ChipToolICDRegistrationDelegate.h"
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/BytesToHex.h>
+
+#include <cstdint>
+
+void ChipToolICDRegistrationDelegate::SetCheckInNodeId(chip::NodeId checkInNodeId)
+{
+    mCheckInNodeId = checkInNodeId;
+}
+
+void ChipToolICDRegistrationDelegate::GenerateSymmetricKey(
+    chip::NodeId device, chip::Callback::Callback<OnSymmetricKeyGenerationCompleted> * onCompletion)
+{
+    ChipLogProgress(chipTool, "Generate symmetric Key for Node ID " ChipLogFormatX64, ChipLogValueX64(device));
+    chip::DeviceLayer::SystemLayer().ScheduleLambda([this, onCompletion] {
```

Comment:
This code is not using onCompletion in a way that's memory-safe.  Our examples should be doing the right thing.